### PR TITLE
Nested Lookup

### PR DESCRIPTION
**Pull request by solomonhawk**
Addresses #1 

This PR adds a few neat features. It adds a few syntax variations for retrieving translations.

``` js
let translations = { foo: '1', bar: '2', baz: { qux: 'lulz' } }
let t = reword(translations)

// Finds the `foo` property in the translations object
t('foo') 
// => '1'

// Finds both the `foo` and `bar` properties in the translations object
t('foo', 'bar') 
// => ['1', '2']

// Nice for retrieving several translations at once. Nice w/ array destructuring
let [ one, two ] = t('foo', 'bar')
// one => '1'
// two => '2'

// Finds the nested property at `translations.baz.qux`
t(['baz', 'qux']) 
// => 'lulz'

// Same as above
t('baz.qux') 
// => 'lulz'
```

### CODE REVIEW

File: src/__tests__/reword.test.js [hunk 1]
```diff
@@ -22,12 +48,15 @@ describe('reword', function() {
...
   describe('when given an unknown string', function() {
 
-    let spy  = sinon.spy()
-    let stub = sinon.stub(console, 'warn', spy)
+    it('calls the notFound handler if present', function() {
+      let spy = sinon.spy()
+      let t = reword(translations, spy)
+      let result = t('not found')
+      expect(spy.calledWith('not found'))
+    })
```
Comment (nhunzaker) [inline src/__tests__/reword.test.js:56]: Nitpick, but `spy` here is really a `stub`.

**Comment by nhunzaker** [comment]
:+1:

File: src/__tests__/reword.test.js (hunk 1 above)
Comment (solomonhawk) [inline src/__tests__/reword.test.js:56]: True true.

**Comment by cwmanning** [comment]
:+1:

**Pull request merged**